lib/sql-string.js

```diff
@@ -35,6 +35,11 @@ function escape(val, timeZone, dialect, format) {
     break;
   }
 
+  // moment
+  if (val.toDate) {
```

Comment:
We can't claim that any thing with `toDate` will follow our specifications, murphy's law is real :) Thus sticking with `isMoment` check with `isValid` is better solution to achieve stability